orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/MonitoredDeployStrategy.groovy

```diff
@@ -421,12 +410,74 @@ class MonitoredDeployStrategy implements Strategy {
     return stages
   }
 
+  List<Stage> composeRollbackStages(Stage parent) {
+    CreateServerGroupStage.StageData stageData = parent.mapTo(CreateServerGroupStage.StageData)
+    MonitoredDeployStageData monitoredDeployStageData = parent.mapTo(MonitoredDeployStageData)
+    String deployedServerGroupName = stageData.getServerGroup()
+
+    // Does the user want an automatic rollback?
+    if ((monitoredDeployStageData.failureActions.rollback != FailureActions.RollbackType.Automatic) &&
+      (monitoredDeployStageData.failureActions.rollback != FailureActions.RollbackType.Manual)) {
+      log.warn("Not performing automatic rollback on failed deploy of ${deployedServerGroupName ?: '<NO SERVER GROUP CREATED>'} because no rollback was requested by user in pipeline config")
```

Comment:
Is this even a valid scenario? where the user can specify he doesn't want rollback?

Reply:
yes, that's the default for all strategies (except for Monitored - we we are opting users in by default and they can optout)